shared/util.go

```diff
@@ -276,3 +289,73 @@ func Retry(f func() error, attempts uint) error {
 
 	return err
 }
+
+// ParseCompression extracts the compression method and level (if any) from the
+// compression flag.
+func ParseCompression(compression string) (string, *int, error) {
+	levelRegex := regexp.MustCompile(`^([\w]+)-(\d{1,2})$`)
+	match := levelRegex.FindStringSubmatch(compression)
+	if match != nil {
+		compression = match[1]
+		level, err := strconv.Atoi(match[2])
+		if err != nil {
+			return "", nil, err
+		}
+
+		switch compression {
+		case "zstd":
+			if 1 <= level && level <= 22 {
+				return compression, &level, nil
+			}
+		case "bzip2", "gzip", "lzop":
```

Comment:
We should allow `lzo` here as well, and return `lzop` in this case as that's what `tar` accepts.